File: leetcode/1207-Unique-Number-of-Occurrences.rs
use std::collections::HashMap;

impl Solution {
    pub fn unique_occurrences(arr: Vec<i32>) -> bool {
        let mut count_dict = HashMap::new();

        for num in arr {
            *count_dict.entry(num).or_insert(0) += 1;
        }

        let values_num = count_dict.len();
        let set_num = count_dict.values().cloned().collect::<std::collections::HashSet<_>>().len();

        if values_num == set_num {
            return true;
        } else {
            return false;
        }

        



        return true;
    }
}

File: leetcode/1207-Unique-Number-of-Occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        count_dict = {}
        for num in arr:
            if num not in count_dict:
                count_dict[num] = 1
            else:
                count_dict[num] += 1

        if len(count_dict.values()) == len(list(set(count_dict.values()))):
            return True
        return False
        